### ISSUE TITLE
Install problem in conda

### ISSUE THREAD

**Issue by liubingdong** [bxlab/metaWRAP#269]
I followed the install procedure, but the conda looks like frozen  at the step 'Solving environment' for hours.

The net works well and I do not know how to fix it.
should I just wait for more time or do something to make it fast?
And ,is there any other channel to install it instead of conda?

Any help is welcome , and thks.

**Comment by liubingdong** [comment]
OK, I finally solve the problem.
Just follow `conda install -y -c ursky metawrap-mg=1.2.4`  and nohub it.
It almost took 2 days to start the installation.
Maybe it could help somebody like me.

**Comment by bph6212ln** [comment]
I'm running into a similar problem rendering metwrap uninstallable with conda. (When installing each dependency individually, It seems to be getting hung up interminably at r-ggplot2.) You write above that it took two days to start the installation. Does that mean you issued this command "conda install -y -c ursky metawrap-mg=1.2.4" and then it remained in the 'Solving Environment' stage for 2 days before it started installing packages?

**Comment by ursky** [comment]
Two days is quite extreme. Try this command instead: `conda create --name metawrap-env --channel ursky --channel bioconda --channel conda-forge metawrap-mg=1.3` so at least it doesn't have to resolve the existing packages in you current env.

**Comment by liubingdong** [comment]
> I'm running into a similar problem rendering metwrap uninstallable with conda. (When installing each dependency individually, It seems to be getting hung up interminably at r-ggplot2.) You write above that it took two days to start the installation. Does that mean you issued this command "conda install -y -c ursky metawrap-mg=1.2.4" and then it remained in the 'Solving Environment' stage for 2 days before it started installing packages?

Yes, it took almost two days to solve Environment. I also tried a new env to install metaWRAP, it is still slow as before. In my opinion, this issue probably is associated with your net.  Hope you work well.



> Two days is quite extreme. Try this command instead: `conda create --name metawrap-env --channel ursky --channel bioconda --channel conda-forge metawrap-mg=1.3` so at least it doesn't have to resolve the existing packages in you current env.

I am glad

> Two days is quite extreme. Try this command instead: `conda create --name metawrap-env --channel ursky --channel bioconda --channel conda-forge metawrap-mg=1.3` so at least it doesn't have to resolve the existing packages in you current env.

Glad to see the new version and your great job help us a lot in this area. Thank you!

**Comment by bph6212ln** [comment]
Thanks for the response, liubingdong. If by 'net' you mean my network connectivity, it is definitely good.

Ursky, thanks for your response too. Your solution above seems to get me a lot further than the 'Best (manual) installation' instructions in the project README.md but it is still failing. After getting through the "Executing Transaction" stage and printing out information about managing the Krona databases it prints:

failed
ERROR conda.core.link:_execute(502): An error occurred while installing package 'conda-forge::backports-1.0-py_2'.
OSError(2, 'No such file or directory')
Attempting to roll back.

Rolling back transaction: done

OSError(2, 'No such file or directory')

The env directory is essentially empty, containing only a bunch of nested subdirectories.

**Comment by ursky** [comment]
Not sure what else to say - it works well on all servers ive worked on so far. My suggestion is to make a clean env, install metawrap only (i.e the stuff from this github), and then install only the software you need for the module you are intending to use (as outlined in `https://github.com/bxlab/metaWRAP/blob/master/installation/dependancies.md`). You might even need to make separate envs for different modules if they really conflict. Again though, Ive never encountered the above error - this is just a workaround for you. Good luck!

**Comment by bph6212ln** [comment]
There must have been something stale in the Conda installation on the otherwise clean CentOS+Anaconda2 Jetstream VM I was using, (I think I should have tried updating everything beforehand). I created a new base CentoOS 7 VM, installed a new copy of Miniconda2 and re-ran the one-line install command you gave above and it succeeded. Thanks for your help and advice!

**Comment by ys117vt** [comment]
Hi ursky @ursky , I am having similar issue here as I 1) failed the best (manual) installation with frozen Solving environment; 2) failed the express installation/bioconda installation at executing transaction step, which warned that Python 2.7 file can not be located in the bin folder. My conda version is 4.5.4 and I have confirmed that channels/priority. Any other recommendation what I should try before I reinstall the Anaconda? Thanks!

**Comment by ZQYYRA** [comment]
A same thing occurred when I install metawrap-mg, the version is 1.3.2, when I do it as" https://[github.com](https://github.com/bxlab/metaWRAP/blob/master/installation/metawrap_installation.md)/bxlab/metaWRAP/blob/master/installation/metawrap_installation.md" described. the conda is still solving environment. and i sloving this problem by the following steps:
 1.delete conda( I installed miniconda3 in my servies)  in my servies cleanly and in ~/.bashrc file is also delete some information about conda. 
2.next is reinstall miniconda3
3. conda install -y mamba
4. mamba create -y -n metawrap-env
5. conda activate metawrap-env
6. conda config --add channels defaults
conda config --add channels conda-forge
conda config --add channels bioconda
conda config --add channels ursky
7.mamba install --only-deps -c ursky metawrap-mg
use mamba is more fast than conda, it just cost me about 5 minutes when I do step7 to install metawrap-mg
i hope it is helpful for us who have the same problem to install metawrap-env
